packages/init/src/steps/starters/documents.ts

```diff
@@ -0,0 +1,114 @@
+import path from "path";
+import fs from "fs";
+import axios, { AxiosError } from "axios";
+
+import * as t from "io-ts";
+import { getOrElseW } from "fp-ts/Either";
+import { handleErrors } from "./communication";
+import { logs } from "../../utils";
+
+const SignatureFileReader = t.type({
+  signature: t.string,
+});
+
+type SignatureFile = t.TypeOf<typeof SignatureFileReader>;
+
+export async function readSignatureFile(cwd: string): Promise<SignatureFile> {
+  const pathToFile = path.join(cwd, "documents", "index.json");
+  return fs.promises.readFile(pathToFile, "utf-8").then((res) => {
+    const data = JSON.parse(res) as unknown;
+    return getOrElseW(() => {
+      throw new Error("Unable to read document signature file");
+    })(SignatureFileReader.decode(data));
+  });
+}
+
+async function lsdir(dir: string): Promise<Array<string>> {
+  return fs.promises.readdir(dir).then((dirs) => {
+    return dirs
+      .filter((name) => fs.statSync(path.join(dir, name)).isDirectory())
```

Comment:
can we make this async with 
const directories = Promise.all([dirs.filter(async => ....]) ?